src/evict/evict_page.c

```diff
@@ -575,6 +575,20 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     if (!modified)
         return (0);
 
+    /*
+     * If we trying to evict a dirty page that does not belong to history store(HS) and checkpoint
+     * is processing HS file, then avoid evicting the dirty non-HS page for now if the cache is
+     * already dominated by dirty HS content.
+     *
+     * Evicting a non-HS dirty page can generate even more HS content. As we can not evict HS pages
+     * while checkpoint is operating on the HS file, we can end up in a situation where we exceed
+     * the cache size limits.
+     */
+    if (modified && conn->txn_global.checkpoint_running_hs && !WT_IS_HS(btree->dhandle) &&
+      __wt_cache_hs_dirty(session) && __wt_cache_full(session)) {
+        WT_STAT_CONN_INCR(session, cache_eviction_checkpoint_hs);
```

Comment:
Rename cache_eviction_checkpoint_hs to cache_eviction_blocked_checkpoint_hs OR some other name that has word "blocked" in it.

Reply:
Fixed.